src/Support/SupportData.php

```diff
@@ -0,0 +1,930 @@
+				'contents'   => '',
+			];
+		}
+
+		$file        = file( $error_log_path );
+		$max_lines   = max( 0, count( $file ) - 200 );
+		$file_length = count( $file );
+		$contents    = [];
+
+		for ( $i = $max_lines; $i < $file_length; $i ++ ) {
+			if ( ! empty( $file[ $i ] ) ) {
+				$contents[] = sanitize_text_field( $file[ $i ] );
+			}
+		}
+
+		return [
+			'log_errors' => ini_get( 'log_errors' ),
+			'contents'   => implode( "\n", $contents ),
+		];
+	}
+
+	/**
+	 * To get plugin information by plugin file.
+	 *
+	 * @since 2.2
+	 *
+	 * @param string $plugin_file Plugin file.
+	 *
+	 * @return array Plugin detail.
+	 */
+	public static function normalize_plugin_info( $plugin_file ) {
+
+		$absolute_plugin_file = WP_PLUGIN_DIR . DIRECTORY_SEPARATOR . $plugin_file;
+		if ( ! file_exists( $absolute_plugin_file ) ) {
+			return [];
+		}
+
+		require_once ABSPATH . '/wp-admin/includes/plugin.php';
+
+		$plugin_data = get_plugin_data( $absolute_plugin_file );
+
+		$slug = explode( '/', $plugin_file );
+		$slug = $slug[0];
+
+		$amp_options        = get_option( 'amp-options' );
+		$suppressed_plugins = ( ! empty( $amp_options['suppressed_plugins'] ) && is_array( $amp_options['suppressed_plugins'] ) ) ? $amp_options['suppressed_plugins'] : [];
+
+		$suppressed_plugin_list = array_keys( $suppressed_plugins );
+
+		if ( empty( $plugin_data['Name'] ) ) {
+			return [];
+		}
+
+		return [
+			'name'              => $plugin_data['Name'],
+			'slug'              => $slug,
+			'plugin_url'        => array_key_exists( 'PluginURI', $plugin_data ) ? $plugin_data['PluginURI'] : '',
+			'version'           => array_key_exists( 'Version', $plugin_data ) ? $plugin_data['Version'] : '',
+			'author'            => array_key_exists( 'AuthorName', $plugin_data ) ? $plugin_data['AuthorName'] : '',
+			'author_url'        => array_key_exists( 'AuthorURI', $plugin_data ) ? $plugin_data['AuthorURI'] : '',
+			'requires_wp'       => array_key_exists( 'RequiresWP', $plugin_data ) ? $plugin_data['RequiresWP'] : '',
+			'requires_php'      => array_key_exists( 'RequiresPHP', $plugin_data ) ? $plugin_data['RequiresPHP'] : '',
+			'is_active'         => is_plugin_active( $plugin_file ),
+			'is_network_active' => is_plugin_active_for_network( $plugin_file ),
+			'is_suppressed'     => in_array( $slug, $suppressed_plugin_list, true ) ? $suppressed_plugins[ $slug ]['last_version'] : '',
+		];
+
+	}
+
+	/**
+	 * To normalize theme information.
+	 *
+	 * @since 2.2
+	 *
+	 * @param \WP_Theme $theme_object Theme object.
+	 *
+	 * @return array Normalize theme information.
+	 */
+	public static function normalize_theme_info( $theme_object ) {
+
+		if ( empty( $theme_object ) || ! is_a( $theme_object, 'WP_Theme' ) ) {
+			return [];
+		}
+
+		$active_theme      = wp_get_theme();
+		$active_theme_slug = '';
+		$parent_theme      = '';
+
+		if ( ! empty( $active_theme ) && is_a( $active_theme, 'WP_Theme' ) ) {
+			$active_theme_slug = $active_theme->get_stylesheet();
+		}
```

Comment:
This condition is not needed as-is because `wp_get_theme()` _always_ returns a `WP_Theme` object. In the case of errors, you can check `$active_theme->errors()`.